src/Tempest/Http/src/RouteDiscovery.php

```diff
@@ -24,21 +28,31 @@ public function discover(ClassReflector $class): void
                 continue;
             }
 
-            $this->routeConfig->addRoute($method, $routeAttribute);
+            $routeAttribute->setHandler($method);
```

Comment:
Moved the handler setting from the RouteConfig to the Discovery .

It seemed really odd to me to move state mutations (setHandler) down (more method calls down) instead of up (more to the surface). I like to have to think as less as possible about the different variations a object has to be and at what time (I really like to use immutable values for this reason).